arbnode/inbox_tracker.go

```diff
@@ -780,3 +780,74 @@ func (t *InboxTracker) ReorgBatchesTo(count uint64) error {
 	log.Info("InboxTracker", "SequencerBatchCount", count)
 	return t.txStreamer.ReorgToAndEndBatch(dbBatch, prevBatchMeta.MessageCount)
 }
+
+func (t *InboxTracker) ReorgMessagesTo(messageCount arbutil.MessageIndex) error {
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+
+	var delayedRead uint64
+	if messageCount > 0 {
+		prevMessage, err := t.txStreamer.GetMessage(messageCount - 1)
+		if errors.Is(err, AccumulatorNotFoundErr) {
+			return errors.New("attempted to reorg to future message count")
```

Comment:
could also be a past pruned message.

Reply:
(sometime we might want to re-read pruned messages for such a case.. for now erroring out is probably good enough)